### PR TITLE
feat: implement focus visible

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description
We need to use `:focus-visible` instead of `:focus`: to allow keyboard navigation to still use visible focus, but remove instances in which pointer interaction shows focus indication.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [ ] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [ ] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
 - [x] Request a review from Brad Paugh, David Becher, or Drew Chandler.

### CODE REVIEW

File: lib/build/less/utilities/internals.less [hunk 1]
```diff
@@ -141,6 +153,9 @@
       .f\:@{className}:focus-within {
         @mixin();
       }
+      .f\:@{className}:focus-visible {
+        @mixin();
+      }
```
Comment (braddialpad) [inline lib/build/less/utilities/internals.less]: this should be .fv: right?

Comment (juliodialpad) [reply lib/build/less/utilities/internals.less]: Right, fixed it good catch, thanks!

**Review by braddialpad — APPROVED** [review]

**Review by nikitadialpad — APPROVED** [review]

File: lib/build/less/components/link.less [hunk 2]
```diff
@@ -41,6 +41,17 @@
...
     &:focus {
+      border-radius: 4px;
+      outline: none;
+      box-shadow: 0 0 0 var(--su2) var(--focus-ring);
+    }
+
+    &:focus:not(:focus-visible) {
+      box-shadow: none;
+    }
+
```
Comment (nikitadialpad) [inline lib/build/less/components/link.less:52]: there're two empty lines

**Comment by nikitadialpad** [comment]
seems corresponding polyfills should be add separately, to not forget about that are we going to add some documentation for this case?

File: lib/build/less/components/link.less (hunk 2 above)
Comment (juliodialpad) [reply lib/build/less/components/link.less:52]: Thanks, fixed!

**Comment by juliodialpad** [comment]
> seems corresponding polyfills should be add separately, to not forget about that are we going to add some documentation for this case?

Yeah, we have a ticket to add focus-visible and polyfill documentation to dialpad.design, so it's ok.

**Pull request merged**